packages/autorest.python/autorest/codegen/__init__.py

```diff
@@ -222,6 +271,10 @@ def process(self) -> bool:
             self.remove_cloud_errors(yaml_data)
 
         code_model = CodeModel(yaml_data=yaml_data, options=options)
+        if options["unbranded"] and any(
```

Comment:
we should put this here https://github.com/Azure/autorest.python/blob/a439cc6f46d990daaf3b5c047813be71572fa8fc/packages/autorest.python/autorest/codegen/__init__.py#L23

Then, we make `options_retriever` an instance property on `CodeGenerator`, move `validate_code_options` onto `CodeGenerator`, and call `self.options_retriever` in both

Reply:
done